xdsl/dialects/arm/register.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import abc
+from collections.abc import Sequence
+
+from xdsl.backend.register_type import RegisterType
+from xdsl.ir import Attribute
+from xdsl.irdl import irdl_attr_definition
+from xdsl.parser import AttrParser
+
+
+class ARMRegisterType(RegisterType, abc.ABC):
+    """
+    The abstract class for all ARM register types.
+    """
+
+    @classmethod
+    def parse_parameters(cls, parser: AttrParser) -> Sequence[Attribute]:
+        if parser.parse_optional_punctuation("<") is not None:
+            name = parser.parse_identifier()
+            parser.parse_punctuation(">")
+            if not name.startswith("x"):  # only including x regs for now
+                assert name in cls.abi_index_by_name(), f"{name}"
+        else:
+            name = ""
+        return cls._parameters_from_spelling(name)
+
+    def verify(self) -> None:
+        return
```

Comment:
Don't think there's any reason to have an empty verify

Reply:
We played around with this locally, and our verify-diangostics runs after the parsing, so it's not trivial to do here. I also have a feeling that raising an error here might be a bit premature, and that we would probably benefit from building more infrastructure here to see how we deal with the weirdnesses of ARM registers before doing the verification.